astacus/coordinator/plugins/clickhouse/steps.py

```diff
@@ -151,6 +153,28 @@ async def run_step(self, cluster: Cluster, context: StepsContext) -> Sequence[Ac
         return access_entities
 
 
+@dataclasses.dataclass
+class RetrieveUDFsStep(Step[Sequence[UDF]]):
+    zookeeper_client: ZooKeeperClient
+    replicated_user_defined_zookeeper_path: str | None
+
+    async def run_step(self, cluster: Cluster, context: StepsContext) -> Sequence[UDF]:
+        if self.replicated_user_defined_zookeeper_path is None:
+            return []
+
+        udfs: list[UDF] = []
+        async with self.zookeeper_client.connect() as connection:
+            change_watch = ChangeWatch()
+            children = await connection.get_children(self.replicated_user_defined_zookeeper_path, watch=change_watch)
```

Comment:
Can `/clickhouse/udfs` be missing if the user has not created any function yet ? (This should probably handle `NoNodeError` if that's not the case)